skyportal/handlers/api/spectrum.py

```diff
@@ -44,62 +43,67 @@
-                    raise AccessError(
-                        f'Invalid {model_class.__name__} IDs field ("{id_list}"); '
-                        f'Not all {model_class.__name__} IDs are valid/accessible'
-                    )
-                validated_ids.append(id)
-        except ValueError:
-            raise ValueError(
-                f'Invalid {model_class.__name__} IDs field ("{id_list}"); '
-                f'Could not parse all elements to integers'
-            )
-
-        return validated_ids
+def parse_id_list(id_list, model_class, session):
+    """
+    Return a list of integer IDs from the comma separated
+    string of IDs given by the query argument, and the
+    model/table to be queried.
+
+    Parameters
+    ----------
+    id_list: string
+        Comma separated list of integer values.
+    model_class: class
+        A skyportal data model class, e.g., Group, Instrument.
+    session: sqlalchemy.Session
+        Database session for this transaction
+    """
+
+    if id_list is None:
+        return  # silently pass through any None values
+
+    try:
+        accessible_rows = session.scalars(
+            model_class.select(session.user_or_token)
+        ).all()
+        validated_ids = []
+        for id in id_list.split(','):
+            id = int(id)
+            if id not in [row.id for row in accessible_rows]:
+                raise AccessError(
+                    f'Invalid {model_class.__name__} IDs field ("{id_list}"); '
+                    f'Not all {model_class.__name__} IDs are valid/accessible'
+                )
+            validated_ids.append(id)
+    except ValueError:
+        raise ValueError(
+            f'Invalid {model_class.__name__} IDs field ("{id_list}"); '
+            f'Could not parse all elements to integers'
+        )
 
-    @staticmethod
-    def parse_string_list(str_list):
-        """
-        Parse a string that is either a single value,
-        or a comma separated list of values.
-        Returns a list of strings in either case.
-        If input is an empty string returns an
-        empty list.
-        """
-        if isinstance(str_list, str):
-            if len(str_list) == 0:
-                return []
-            elif "," in str_list:
-                return [c.strip() for c in str_list.split(",")]
-            else:
-                return [str_list.strip()]
+    return validated_ids
+
+
+def parse_string_list(str_list):
+    """
+    Parse a string that is either a single value,
+    or a comma separated list of values.
+    Returns a list of strings in either case.
+    If input is an empty string returns an
+    empty list.
+    """
+    if isinstance(str_list, str):
+        if len(str_list) == 0:
+            return []
+        elif "," in str_list:
+            return [c.strip() for c in str_list.split(",")]
         else:
-            raise TypeError('Must input a string!')
+            return [str_list.strip()]
```

Comment:
I think you get the same result by removing the last `else` and replacing the `elif` with an `else`.

Reply:
I had to stare at this comment for a while to try to figure out why it was the case ;)